README.md

```diff
@@ -53,6 +53,125 @@ The SDK assemblies are strongly named which requires consumers of the SDK to rec
 
+## Making SDK Changes
+
+### DevConfig Files
+
+DevConfig files are required for all developer .NET SDK branches being merged into the system. These files control versioning for the change, changelog messages, which services are being updated, and other controls. The purpose of bringing developer configuration information into each developer's branch is so that any configuration and changelog messages the developer intends to release is reviewed by other developers reviewing their PR.
+
+DevConfig files are JSON files placed in the `./generator/.DevConfigs` folder. Each file must have a unique name to avoid merge conflicts. The file contains information about core and service changes, which helps the build system process and properly version bump the NuGet packages.
+
+#### DevConfig File Structure
+
+A DevConfig file can contain the following sections:
+
+**Core Section**:
+Required for any change to the Core component of the SDK.
+
+```json
+{
+  "core": {
+    "changeLogMessages": [
+      "Message 1",
+      "Message 2"
+    ],
+    "type": "patch",
+    "updateMinimum": true
+  }
+}
+```
+
+- `changeLogMessages`: (Required) An array of messages to include in the changelog under the core section.
+- `type`: (Required) The part of the version string to increment. Valid values are `minor` and `patch`.
+- `updateMinimum`: (Required) Whether to update the minimum core version for all services.
+- `backwardIncompatibilitiesToIgnore`: (Optional) An array of strings for the type of validation errors to ignore.
+
+**Service Section**:
+Required for any change to a specific service.
+
+```json
+{
+  "services": [
+    {
+      "serviceName": "S3Control",
+      "type": "patch",
+      "changeLogMessages": [
+        "Fixed a bug causing a stack overflow when using the adaptive retry policy."
+      ]
+    },
+    {
+      "serviceName": "S3",
+      "type": "patch",
+      "changeLogMessages": [
+        "Fixed a bug where the bucketName was being marshalled incorrectly in operation xyz."
+      ]
+    }
+  ]
+}
+```
+
+- `serviceName`: (Required) Name of the service without the "AWSSDK." prefix.
+- `type`: (Required) The part of the version string to increment. Valid values are `minor` and `patch`.
+- `changeLogMessages`: (Required) An array of messages to include in the changelog.
+
+**Other Special Sections**:
+
+- `overrideVersion`: Allows specifying a specific version to use for all services in the release.
+
+```json
+{
+  "services": [...],
+  "core": {...},
+  "overrideVersion": "3.7.300.0"
+}
+```
+
+#### Creating a DevConfig File
+
+You can use the provided script to generate a DevConfig file:
+
+```
+./buildtools/add-devconfig.bat
```

Comment:
Call out the bat file for windows or `./buildtools/add-devconfig.sh` for mac/linux